Reply/app/src/main/java/com/materialstudies/reply/ui/MainActivity.kt

```diff
@@ -165,34 +117,85 @@ class MainActivity : AppCompatActivity(),
 
-    private fun navigateToCompose() {
+    fun navigateToSearch() {
         currentNavigationFragment?.apply {
-            exitTransition = MaterialElevationScale(false).apply {
+            exitTransition = MaterialSharedAxis(MaterialSharedAxis.Z, true).apply {
                 duration = resources.getInteger(R.integer.reply_motion_duration_large).toLong()
             }
-            reenterTransition = MaterialElevationScale(true).apply {
+            reenterTransition = MaterialSharedAxis(MaterialSharedAxis.Z, false).apply {
                 duration = resources.getInteger(R.integer.reply_motion_duration_large).toLong()
             }
         }
-        val directions = ComposeFragmentDirections.actionGlobalComposeFragment(currentEmailId)
+        val directions = SearchFragmentDirections.actionGlobalSearchFragment()
         findNavController(R.id.nav_host_fragment).navigate(directions)
     }
 
-    private fun navigateToSearch() {
+    private fun showDarkThemeMenu() {
+        MenuBottomSheetDialogFragment
+            .newInstance(R.menu.dark_theme_bottom_sheet_menu)
+            .show(supportFragmentManager, null)
+    }
+
+    private fun navigateToCompose() {
         currentNavigationFragment?.apply {
-            exitTransition = MaterialSharedAxis(MaterialSharedAxis.Z, true).apply {
+            exitTransition = MaterialElevationScale(false).apply {
                 duration = resources.getInteger(R.integer.reply_motion_duration_large).toLong()
             }
-            reenterTransition = MaterialSharedAxis(MaterialSharedAxis.Z, false).apply {
+            reenterTransition = MaterialElevationScale(true).apply {
                 duration = resources.getInteger(R.integer.reply_motion_duration_large).toLong()
             }
         }
-        val directions = SearchFragmentDirections.actionGlobalSearchFragment()
+        val directions = ComposeFragmentDirections.actionGlobalComposeFragment(currentEmailId)
         findNavController(R.id.nav_host_fragment).navigate(directions)
     }
 
     private fun applyBackgroundColor() {
         val surfaceColor5 = SurfaceColors.SURFACE_5.getColor(this)
         window.decorView.setBackgroundColor(surfaceColor5)
     }
+
+    private fun setUpBottomNavigationAndFab() {
+        // Wrap binding.run to ensure ContentViewBindingDelegate is calling this Activity's
+        // setContentView before accessing views
+        binding.run {
+            findNavController(R.id.nav_host_fragment).addOnDestinationChangedListener(
+                this@MainActivity
+            )
+        }
+
+        // Set a custom animation for showing and hiding the FAB
+        binding.fab.apply {
+            setShowMotionSpecResource(R.animator.fab_show)
+            setHideMotionSpecResource(R.animator.fab_hide)
+            setOnClickListener {
+                navigateToCompose()
+            }
+        }
+        setUpBottomNavClickListeners()
+    }
+
+    private fun setUpBottomNavClickListeners() {
```

Comment:
I wouldn't break this out into it's own method, but maybe that's personal preference.

Reply:
Don't really have a preference so this is okay.